torch/csrc/distributed/autograd/utils.cpp

```diff
@@ -52,6 +59,33 @@ DistAutogradContext* addRecvRpcBackward(
   return nullptr;
 }
 
+std::shared_ptr<FutureMessage> sendMessageWithAutograd(
```

Comment:
it looks like this function doesn't always send a message with autograd (if there is no valid context). If we want to name this function `sendMessageWithAutograd `, should we refactor to remove this check and do the check before invoking `sendMessageWithAutograd`?